### PR TITLE
Update connector service on-prem doc

### PR DESCRIPTION
**Pull request by wangch079**
Update the markdown document to reflect the latest UI changes in 8.8

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [ ] this PR links to all relevant github issues that it fixes or partially addresses
- [ ] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: docs/CONFIG.md [hunk 1]
```diff
@@ -59,14 +59,21 @@ When you have an Enterprise Search deployment on Elastic Cloud post 8.5.0, the c
...
+1. Go to Kibana, _Enterprise Search_ > _Create an Elasticsearch index_ > _Connector_. Choose `Customized connector` and continue to create an index.
 2. Create an API key to work with the connector. It should be done using the `Generate API key` button under `Configuration` tab.
 3. Configure your connector service application. You need to configure the followings fields, and leave the rest as default.
    1. `elasticsearch.host`: Configure this to the Elasticsearch endpoint.
    2. `elasticsearch.api_key`: Configure the API key generated in step 2. Make sure `elasticsearch.username` is not configured.
    3. `connector_id`: You can find the `connector_id` in step 3 `Deploy a connector` under `Configuration` tab in Kibana.
    4. `service_type`: Configure it to the service type of your new connector.
+   5. Leave `native_service_types` empty, unless you want to sync any native connectors with this connector service instance. In such case, make sure the API key configured in step 2 has the proper privileges, or you can option to use basic auth (`username` and `password`) with proper privileges.
```
Comment (seanstory) [inline docs/CONFIG.md:69]:
```suggestion
   5. Leave `native_service_types` empty, unless you want to sync any native connectors with this connector service instance. In such a case, make sure the API key configured in step 2 has the proper privileges, or you can choose to use basic auth (`username` and `password`) with the proper privileges.
```

File: docs/CONFIG.md [hunk 2]
```diff
@@ -59,14 +59,21 @@ When you have an Enterprise Search deployment on Elastic Cloud post 8.5.0, the c
...
 4. Run the connector service application with
     ```shell
     make run
     ```
+
+### Run the connector service for a converted native connector
+
+1. Follow the instructions in [Run the connector service in native mode](#run-the-connector-service-in-native-mode) to create a native connector.
+2. Go to `Configuration` tab and convert it to a customized connector via `Convert connector`.
```
Comment (seanstory) [inline docs/CONFIG.md:78]:
```suggestion
2. Go to the `Configuration` tab and convert it to a customized connector via `Convert connector`.
```

**Review by seanstory — DISMISSED** [review]
Two small phrasing nits. But LGTM!

File: docs/CONFIG.md [hunk 3]
```diff
@@ -59,14 +59,21 @@ When you have an Enterprise Search deployment on Elastic Cloud post 8.5.0, the c
...
 4. Run the connector service application with
     ```shell
     make run
     ```
+
+### Run the connector service for a converted native connector
+
+1. Follow the instructions in [Run the connector service in native mode](#run-the-connector-service-in-native-mode) to create a native connector.
+2. Go to the `Configuration` tab and convert it to a customized connector via `Convert connector`.
+3. Follow the instructions in [Run the connector service for a custom connector](#run-the-connector-service-for-a-custom-connector) to run the connector service.
```
Comment (ppf2) [inline docs/CONFIG.md:79]: Nice, thanks for adding this section!  Are we saying that regardless of whether they are customizing the native connector or leaving it unmodified after conversion, they must treat it like a custom connector?  Or should we split this up into two sections, 1 for converted native connector (unmodified) v.s. converted native connector (customized)?

**Comment by ppf2** [comment]
Addition to the changes here, i think we also want to update the https://github.com/elastic/connectors-python/blob/8.8/docs/CONFIG.md#run-the-connector-service-in-native-mode section to indicate that this is section also applies if they are try to run a connector client (i.e. it's not just for running native connectors on-prem).

File: docs/CONFIG.md (hunk 3 above)
Comment (wangch079) [reply docs/CONFIG.md:79]: Sorry I have totally overlooked this issue.

>Are we saying that regardless of whether they are customizing the native connector or leaving it unmodified after conversion, they must treat it like a custom connector?

Yes, as long as it's not a native connector, it should be treated as a custom connector and only run by an on-prem connector service.

With the [work](https://github.com/elastic/connectors-python/pull/1063) done by Sean, I will remove the section to run connector service on-prem in native mode.

**Comment by wangch079** [comment]
>Addition to the changes here, i think we also want to update the https://github.com/elastic/connectors-python/blob/8.8/docs/CONFIG.md#run-the-connector-service-in-native-mode section to indicate that this is section also applies if they are try to run a connector client (i.e. it's not just for running native connectors on-prem).

This section will be removed as mentioned here: https://github.com/elastic/connectors-python/pull/944#discussion_r1230583083

connector client needs to be treated as a custom connector, i.e. configure `connector_id` and `service_type` in yaml and run it on-prem. 

Now it was made even simpler:
1. Native connectors can (or should) only be run on Cloud. It's still possible to run native connectors on-prem, but a deprecation warning will be printed.
2. Connector client or custom connectors can only be run on-prem.

**Comment by wangch079** [comment]
@elastic/ingestion-team , should we remove the section [Running connector service in [native mode](https://www.elastic.co/guide/en/enterprise-search/current/native-connectors.html) in Docker](https://github.com/elastic/connectors-python/blob/main/docs/DOCKER.md#running-connector-service-in-native-mode-in-docker)? Although it's possible but we shouldn't document it. WDYT?

**Comment by wangch079** [comment]
Close this PR as @seanstory updated in a separate PR #1109

**Comment by seanstory** [comment]
🤦 sorry Chenhui, totally forgot that you had this PR too.

**Pull request closed without merging**